model-engine/tests/unit/domain/test_llm_use_cases.py

```diff
@@ -284,6 +284,41 @@ async def test_create_model_endpoint_text_generation_inference_use_case_success(
         )
 
 
+def test_load_model_weights_sub_commands(
+    fake_model_bundle_repository,
+    fake_model_endpoint_service,
+    fake_docker_repository_image_always_exists,
+    fake_model_primitive_gateway,
+    fake_llm_artifact_gateway,
+):
+    fake_model_endpoint_service.model_bundle_repository = fake_model_bundle_repository
+    bundle_use_case = CreateModelBundleV2UseCase(
+        model_bundle_repository=fake_model_bundle_repository,
+        docker_repository=fake_docker_repository_image_always_exists,
+        model_primitive_gateway=fake_model_primitive_gateway,
+    )
+    llm_bundle_use_case = CreateLLMModelBundleV1UseCase(
+        create_model_bundle_use_case=bundle_use_case,
+        model_bundle_repository=fake_model_bundle_repository,
+        llm_artifact_gateway=fake_llm_artifact_gateway,
+        docker_repository=fake_docker_repository_image_always_exists,
+    )
+
+    framework = LLMInferenceFramework.TEXT_GENERATION_INFERENCE
```

Comment:
nit: can you also write a vllm test? It'd be nice to see both working since they test two different branches of conditional logic.